### PR TITLE
Adjusts KapGrenz of lland_control to write selected option to control files instead of derived values

### PR DESCRIPTION
**Pull request by sivogel**
Previously, when saving the control files, for parameter like `KapGrenz` the internally set values were saved and not the option itself. This is now possible (see tyralla musk coefficients), which is why I revised `KapGrenz` and added this functionality.

### CODE REVIEW

File: hydpy/models/lland/lland_control.py [hunk 1]
```diff
@@ -840,35 +842,73 @@ class KapGrenz(parametertools.Parameter):
...
+    If we change the values and they do not comply with the option, the values
+    instead of the option will be returned
+
+    >>> kapgrenz.values = 100.0
+    >>> kapgrenz
+    kapgrenz(100.0)
+
+    If we change the parameter |FK|, on which |KapGrenz| depends, |KapGrenz| will not
+    be updated, but will remain at its old values. The option is no longer returned
+    because it is not valid anymore.
```
Comment (tyralla) [inline hydpy/models/lland/lland_control.py:879]: "." -> ":"

blank line

File: hydpy/models/lland/lland_control.py [hunk 2]
```diff
@@ -840,35 +842,73 @@ class KapGrenz(parametertools.Parameter):
...
+
+    If we change the values and they do not comply with the option, the values
+    instead of the option will be returned
+
+    >>> kapgrenz.values = 100.0
+    >>> kapgrenz
+    kapgrenz(100.0)
+
+    If we change the parameter |FK|, on which |KapGrenz| depends, |KapGrenz| will not
+    be updated, but will remain at its old values. The option is no longer returned
```
Comment (tyralla) [inline hydpy/models/lland/lland_control.py:878]: returned -> printed (?)

File: hydpy/models/lland/lland_control.py [hunk 3]
```diff
@@ -840,35 +842,73 @@ class KapGrenz(parametertools.Parameter):
...
+           [ 60., 120.],
+           [ 90., 180.]])
+    >>> fk(100)
+    >>> kapgrenz
     kapgrenz([[30.0, 60.0],
               [60.0, 120.0],
               [90.0, 180.0]])
 
-    Wrong keyword arguments result in errors like the following:
+    The option will also not be returned if it leads to nan values:
```
Comment (tyralla) [inline hydpy/models/lland/lland_control.py:893]: returned -> printed ("?")
nan -> `nan` (or, better, |numpy.nan|)

blank line

File: hydpy/models/lland/lland_control.py [hunk 4]
```diff
@@ -880,41 +920,50 @@ class KapGrenz(parametertools.Parameter):
...
-                    f"the options `FK`, `0_WMax/10`, and `FK/2_FK`, but "
-                    f"`{kwargs['option']}` is given."
-                ) from None
+        else:
+            self._keywordarguments = parametertools.KeywordArguments(
+                option=kwargs["option"]
+            )
+            self.values = self._get_values_according_to_option()
+
+    def _get_values_according_to_option(self) -> NDArrayFloat:
```
Comment (tyralla) [inline hydpy/models/lland/lland_control.py:936]: NDArrayFloat -> MatrixFloat (a little more precise; will become relevant when we start to type the shape of our Parameter and Sequence_ subclasses.)

File: hydpy/models/lland/lland_control.py [hunk 5]
```diff
@@ -840,35 +842,73 @@ class KapGrenz(parametertools.Parameter):
...
     >>> kapgrenz(option="FK/2_FK")
     >>> kapgrenz
+    kapgrenz(option="FK/2_FK")
+    >>> kapgrenz.values
+    array([[ 30.,  60.],
+           [ 60., 120.],
+           [ 90., 180.]])
+
+    >>> kapgrenz
+    kapgrenz(option="FK/2_FK")
```
[the remark this answers is not in the document]
Comment (sivogel) [reply hydpy/models/lland/lland_control.py]: It was by accident. I now added a new doctest to check if the option is still printed if we set the values manually

File: hydpy/models/lland/lland_control.py [hunk 6]
```diff
@@ -840,35 +842,73 @@ class KapGrenz(parametertools.Parameter):
...
+
+    >>> kapgrenz.values = [[30.0, 60.0], [60.0, 120.0], [90.0, 180.0]]
+    >>> kapgrenz
+    kapgrenz(option="FK/2_FK")
```
[the remark this answers is not in the document]
Comment (sivogel) [reply hydpy/models/lland/lland_control.py:868]: See comment above

File: hydpy/models/lland/lland_control.py (hunk 1 above)
Comment (sivogel) [reply hydpy/models/lland/lland_control.py]: see above

File: hydpy/models/lland/lland_control.py (hunk 2 above)
Comment (sivogel) [reply hydpy/models/lland/lland_control.py]: I always changed returned to printed.

File: hydpy/models/lland/lland_control.py (hunk 3 above)
Comment (sivogel) [reply hydpy/models/lland/lland_control.py]: I corrected to printed and [numpy.nan|

File: hydpy/models/lland/lland_control.py (hunk 4 above)
Comment (sivogel) [reply hydpy/models/lland/lland_control.py]: Done.

**Pull request merged**